src/blog/stylo-another-kind-of-rich-text-editor.md

```diff
@@ -0,0 +1,164 @@
+
+*****
+
+### But Why?
+
+While developing last year the migration of our web editor for slides, [DeckDeckGo](https://deckdeckgo.com), to [DFINITY](https://dfinity.org)’s Internet Computer, we also initialized a new experimental concept to support documents as well.
+
+As a very first test, I wrote and published a blog post on our platform — [Converting SVG To Image (PNG, Webp, etc.) in JavaScript](https://zhx6p-eqaaa-aaaai-abbrq-cai.raw.ic0.app/d/converting-svg-to-image-in-javascript).
+
+Although I was satisfied with the result, I was of course also aware that further enhancements were needed.
+
+That’s why I had the idea to extract the rich text editor capabilities of the core of our application. By limiting the scope of these features, I assumed it would be easier to improve and fix what still needed to be done.
+
+In addition, by publishing the new library as a new open source repo, I was also hoping it would benefit from feedbacks and contributions.
+
+What can I add? I am an engineer, I like to code 😉.
+
+*****
+
+### Features
+
+Not looking to oversell what Stylo can do — or do differently — but here are in my humble opinion a list of cool things it can do:
+
+*****
+
+#### Interactive Design 🎯
+
+Unlike other WYSIWYG editor Stylo has no sticky toolbar. It is inspired by design and user experience of platforms that I like — i.e [Notion](https://www.notion.so/) and [Medium](https://medium.com/).
+
+![](https://cdn-images-1.medium.com/max/1600/1*D-YYgIa5cY0Ya8FAN0Lyiw.gif)
+
+When writing the document, user can use different types of paragraph that can be added through a floating menu. These elements can be simple HTML elements — i.e. h1, h2, code etc. — or more complex block — i.e. any Web Components.
+
+For styling purpose, another toolbar gives the user the ability to modify text — e.g. bold, italic, colors, etc. — and add hyperlinks.
+
+Last but not least, for an optimized experience on desktop, many actions can be performed without the need of the mouse. Opening the list of paragraphs can be triggered hitting “/” and **bold**, *italic* and `mark` are accessible through keyboard shortcuts.
+
+*****
+
+#### Customizable 💪
+
+Conceptually, only one main element — a container — has to be set as editable and all its children are either HTML Elements or Web Components — i.e. no direct text node descendant.
+
+This decision was made for performance reason. In the future we might need to lazy load the content or improve the scrolling experience with some sort of infinite or other virtual scrolling approach. By limiting children to elements, it is possible to assign attributes such as unique ids directly in the DOM. <br>
+
+This brings us to the customization. The list of paragraphs that can be added by the user is fully customizable. One entry — i.e. a plugin — in the list is a text, an icon and a function that adds a new element to the editable container.
```

Comment:
> One entry — i.e. a plugin — in the list is a text, an icon and a function that adds a new element to the editable container.

I'm struggling to understand this. Do you mean that by hitting e.g. `/`, the user can then... add a function to the document?

Reply:
Ok, looking at the snippet below it makes more sense. I think this is the part that threw me off:

> The list of paragraphs that can be added by the user is fully customizable.

I thought you meant "the list of HTML elements in the document", i.e. the paragraphs themselves, whereas you actually mean the type of paragraphs. Correct?

Reply:
rephrased as following, hope it is a bit clearer:

The miscellaneous types of paragraphs that can be added by the user - e.g. block of code, titles, etc. - is fully customizable. One type - i.e. a plugin - is defined with a ​text​, an ​icon​ and a​ function that adds a new element to the document.